cmd/crypto/metadata.go

```diff
@@ -155,12 +174,17 @@ func (s3) ParseMetadata(metadata map[string]string) (keyID string, kmsKey []byte
 	if !ok {
 		return keyID, kmsKey, sealedKey, Error{"The object metadata is missing the internal sealed key for SSE-S3"}
 	}
-	keyID, ok = metadata[S3KMSKeyID]
-	if !ok {
+
+	// There are two possibilites:
+	// - We use a KMS -> There must be a key ID and a KMS data key.
+	// - We use a K/V -> There must be no key ID and no KMS data key.
+	// Otherwise, the metadata is corrupted.
+	keyID, idPresent := metadata[S3KMSKeyID]
+	b64KMSSealedKey, kmsKeyPresent := metadata[S3KMSSealedKey]
```

Comment:
It would be nice to group the detection of whether the object is encrypted in a KV store based setup or an external KMS based setup. I understand that this PR is in preparation towards full-fledged support for encryption key stored in a KV store. We can address this in the PR implementing the support for KV store interface as an alternative to the existing KMS based SSE-S3 support.